[FIX] hr_recruitment: keep tags upon candidate creation

vandroogenbd opened the pull request:
Issue:
When creating an Application, creating a new Candidate removes all tags already in use.

Steps to reproduce:
- Install Recruitment app
- Create a new Application
- Add a tag
- Create a new Candidate

Cause:
The tags of an Application are stored in the 'categ_ids' variable. This variable is changed by the '_compute_categ_ids' method, which sets them to those of the Candidate. Since categ_ids is empty for new Candidates, as of right now, this method is effectively removing all tags for new Candidates.

Desired Behaviour:
Adding a Candidate should add it's tags to the Applicant's.
Removing a Candidate should not change the Applicant's tags.

Ticket:
opw-4351837

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/189339.png)](https://mergebot.odoo.com/odoo/odoo/pull/189339)

abla001 reviewed (commented):
the fix LGTM! but it needs a test :smile:

vandroogenbd commented:
I added some tests and changed the fix a little to keep the behaviour as close to the original one as possible.
Will discuss with PO to know what is the exact intended behaviour.

vandroogenbd commented:
PO greenlit the following behaviour:
1. Adding a Candidate should add it's tags to the Applicant's ones
2. Removing a Candidate should not change the Applicant's tags

abla001 reviewed (approved):
LGTM !

tivisse commented:
@robodoo r+
